lib/src/list_multimap/built_list_multimap.dart

```diff
@@ -170,7 +170,7 @@ class BuiltListMultimap<K, V> {
         throw new ArgumentError('map contained invalid key: ${key}');
       }
 
-      _map[key] = new BuiltList<V>(lookup(key));
+      _map[key as K] = new BuiltList<V>(lookup(key));
```

Comment:
This means we'll do two runtime checks. How about

if (key is K) {
  _map[key] =...
} else {
  throw new ...
}

?